### ISSUE TITLE
Incorrect ContainerPortArgs property in documentation for C#

### ISSUE THREAD

**Issue by konnta0** [pulumi/pulumi-kubernetes#2166]
File: [themes/default/content/registry/packages/kubernetes/api-docs/apps/v1/deployment/_index.md](https://www.pulumi.com/registry/packages/kubernetes/api-docs/apps/v1/deployment/)

In the documentation, the property name that specifies the port for ContainerPortArg is `ContainerPort`, but I think `ContainerPortValue` is correct.
https://www.pulumi.com/registry/packages/kubernetes/api-docs/apps/v1/deployment/#example-usage

incorrect
```cs
new Kubernetes.Types.Inputs.Core.V1.ContainerArgs
{
    Image = "nginx:1.14.2",
    Name = "nginx",
    Ports = 
    {
        new Kubernetes.Types.Inputs.Core.V1.ContainerPortArgs
        {
            ContainerPort = 80,
        },
    },
},
```

correct
```cs
new Kubernetes.Types.Inputs.Core.V1.ContainerArgs
{
    Image = "nginx:1.14.2",
    Name = "nginx",
    Ports = 
    {
        new Kubernetes.Types.Inputs.Core.V1.ContainerPortArgs
        {
            ContainerPortValue = 80,
        },
    },
},
```
https://www.pulumi.com/registry/packages/kubernetes/api-docs/apps/v1/deployment/#containerport
https://github.com/pulumi/pulumi-kubernetes/blame/master/sdk/dotnet/Core/V1/Inputs/ContainerPortArgs.cs#L22

**Comment by viveklak** [comment]
@konnta0 thanks for opening the issue!
Looks like the schema generation is being a bit too cautious here: https://github.com/pulumi/pulumi-kubernetes/blob/fa399ca3a3eae8401de717c67dc351e6932d6d91/provider/pkg/gen/schema.go#L584
i.e. the above special casing should only take effect when the resource kind in question is not of a primitive type. Tightening this would mean that some users' programs would fail to compile and need to be updated to address this.

However, we are also running into a bug with `pulumi convert` used to generate the code you mentioned. I have opened https://github.com/pulumi/pulumi/issues/10671 to track this.

**Comment by viveklak** [comment]
@lblackstone for thoughts on the first part.

**Comment by lblackstone** [comment]
> the above special casing should only take effect when the resource kind in question is not of a primitive type. Tightening this would mean that some users' programs would fail to compile and need to be updated to address this.

I'd vote to fix the incorrect code generation, since this change shouldn't affect resource state. We'd need to bump the minor version of the provider and include a note about the change, but I think that's preferable to docs being out of sync and having problems with YAML conversion.

**Comment by lblackstone** [comment]
Fixed in #2476
